src/sema/function_annotation.rs

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    ast::{ConstructorAnnotation, Diagnostic, Namespace, Type},
+    diagnostics::Diagnostics,
+    expression::{expression, ExprContext, ResolveTo},
+    unused_variable::used_variable,
+    Symtable,
+};
+use crate::Target;
+use solang_parser::pt::{self, CodeLocation};
+
+/// Collect the seeds, bump, payer, and space for constructors. This is a no-op on Substrate/EVM since
+/// there should be no seed or bump annotations permitted on other targets.
+pub fn function_annotations(
+    function_no: usize,
+    annotations: &[&pt::Annotation],
+    symtable: &mut Symtable,
+    context: &ExprContext,
+    ns: &mut Namespace,
+) {
+    let mut diagnostics = Diagnostics::default();
+
+    // On Solana, the seeds and bump for a constructor can be specified using annotations, for example
+    //
+    // @seed(param1)
+    // @seed("fizbaz")
+    // @bump(param2)
+    // constructor(bytes param1, uint8 param2) {}
+
+    let mut parsed_annotations = Vec::new();
+    let mut bump = None;
+    let mut space = None;
+    let mut payer = None;
+
+    for note in annotations {
+        if ns.target == Target::Solana
+            && ns.functions[function_no].ty == pt::FunctionTy::Constructor
+        {
+            match note.id.name.as_str() {
+                "seed" => {
+                    let ty = Type::Slice(Box::new(Type::Bytes(1)));
+                    let loc = note.loc;
+
+                    if let Ok(expr) = expression(
+                        &note.value,
+                        context,
+                        ns,
+                        symtable,
+                        &mut diagnostics,
+                        ResolveTo::Type(&ty),
+                    ) {
+                        if let Ok(expr) = expr.cast(&expr.loc(), &ty, true, ns, &mut diagnostics) {
+                            if let Some(prev) = &bump {
+                                diagnostics.push(Diagnostic::error_with_note(
+                                    *prev,
+                                    "@bump should be after the last @seed".into(),
+                                    loc,
+                                    "location of @seed annotation".into(),
+                                ));
+                            } else {
+                                used_variable(ns, &expr, symtable);
+                                parsed_annotations.push(ConstructorAnnotation::Seed(expr));
+                            }
+                        }
+                    }
+                    continue;
```

Comment:
There is no need to write continue in any of this `match` branches.

Reply:
The `continue` was necessary because of the default diagnostic under it, but now it's re-written without `continue`.